src/ua/griddynamics/Game.java

```diff
@@ -0,0 +1,183 @@
+
+    private final char[][] hiddenField;
+    private final char[][] visibleField;
+
+    public Game(char[][] hiddenField, char[][] visibleField) {
+        this.hiddenField = hiddenField;
+        this.visibleField = visibleField;
+    }
+
+    public void placeMines(Coordinate coordinate, int numberOfMines) {
+        Random random = new Random();
+        int minesPlaced = 0;
+        int userX = coordinate.getX();
+        int userY = coordinate.getY();
+        do {
+            int x = random.nextInt(WIDTH);
+            int y = random.nextInt(HEIGHT);
+            if (hiddenField[y][x] != X && x != userX || y != userY) {
+                hiddenField[y][x] = X;
+                minesPlaced++;
+            }
+        } while (minesPlaced < numberOfMines);
+    }
+
+    public void buildEmptyField() {
+        for (char[] chars : visibleField) {
+            Arrays.fill(chars, POINT);
+        }
+    }
+
+    public String getField(char[][] mineField) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(" |123456789| ");
+        sb.append(System.lineSeparator());
+        sb.append("-|---------|");
+        sb.append(System.lineSeparator());
+        int rowCounter = 1;
+        for (char[] chars : mineField) {
+            sb.append(rowCounter).append("|");
+
+            for (int j = 0; j < chars.length; j++) {
+                sb.append(chars[j]);
+            }
+            sb.append("|");
+            sb.append(System.lineSeparator());
+            rowCounter++;
+        }
+        sb.append("-|---------|");
+        return sb.toString();
+    }
+
+    private int countSurround(int i, int j) {
+        int count = 0;
+        if (i + 1 != HEIGHT && hiddenField[i + 1][j] == X)
+            count++;
+        if (i + 1 != HEIGHT && j + 1 != WIDTH && hiddenField[i + 1][j + 1] == X)
+            count++;
+        if (i - 1 != -1 && j + 1 != WIDTH && hiddenField[i - 1][j + 1] == X)
+            count++;
+        if (i + 1 != HEIGHT && j - 1 != -1 && hiddenField[i + 1][j - 1] == X)
+            count++;
+        if (i - 1 != -1 && j - 1 != -1 && hiddenField[i - 1][j - 1] == X)
+            count++;
+        if (i - 1 != -1 && hiddenField[i - 1][j] == X)
+            count++;
+        if (j + 1 != WIDTH && hiddenField[i][j + 1] == X)
+            count++;
+        if (j - 1 != -1 && hiddenField[i][j - 1] == X)
+            count++;
+        return count;
+    }
+
+    public void numberGenerator() {
+        for (int i = 0; i < HEIGHT; i++) {
+            for (int j = 0; j < WIDTH; j++) {
+                if (hiddenField[i][j] != X) {
+                    hiddenField[i][j] = (char) (countSurround(i, j) + 48);
+                }
+                if (hiddenField[i][j] == '0') {
+                    hiddenField[i][j] = POINT;
+                }
+            }
+        }
+    }
+
+    public void floodFill(int x, int y, boolean[][] visited) {
+        if (x < 0 || x >= 9 || y < 0 || y >= 9) return;
```

Comment:
What is magic number `9`?
Please, use constants

Reply:
Done